scalr/resource_scalr_vcs_provider.go

```diff
@@ -167,6 +198,21 @@ func resourceScalrVcsProviderUpdate(ctx context.Context, d *schema.ResourceData,
 		}
 	}
 
+	if environmentsI, ok := d.GetOk("environments"); ok {
+		environments := environmentsI.(*schema.Set).List()
+		if (len(environments) == 1) && (environments[0].(string) == "*") {
+			options.IsShared = scalr.Bool(true)
+			options.Environments = make([]*scalr.Environment, 0)
+		} else {
+			options.IsShared = scalr.Bool(false)
+			environmentValues := make([]*scalr.Environment, 0)
+			for _, env := range environments {
+				environmentValues = append(environmentValues, &scalr.Environment{ID: env.(string)})
+			}
+			options.Environments = environmentValues
+		}
+	}
```

Comment:
```suggestion
	} else {
		options.IsShared = scalr.Bool(false)
		options.Environments = make([]*scalr.Environment, 0)
	}
```

Reply:
this would introduce BC, currently all configurations create shared providers, with this changes will be planned without change in configuration.